src/main/java/com/microsoft/jfr/FlightRecorderConnection.java

```diff
@@ -143,6 +136,30 @@ public long startRecording(RecordingOptions recordingOptions, RecordingConfigura
         }
     }
 
+    private void setConfiguration(RecordingConfiguration recordingConfiguration, long id) throws OpenDataException, InstanceNotFoundException, MBeanException, ReflectionException, IOException {
+        if (recordingConfiguration == null || recordingConfiguration.getConfiguration() == null) {
+            return;
+        }
+
+        if (recordingConfiguration instanceof RecordingConfiguration.MapConfiguration) {
+            Map<String, String> configuration = (Map) recordingConfiguration.getConfiguration();
+            if (!configuration.isEmpty()) {
```

Comment:
Can it be null?

Reply:
`configuration` can't be null at this point with [this check](https://github.com/jeanbisutti/jfr-streaming/blob/cfcc450b92cec0d51255081232eab3f435596551/src/main/java/com/microsoft/jfr/FlightRecorderConnection.java#L140)